features/202-session-management/use-cases.md

```diff
@@ -0,0 +1,36 @@
+# Use Cases
+
+This document describes non-obvious solution to use cases.
+
+## Revoke all sessions, including current session
+Developer can revoke all other sessions, then revoke current session.
+
+```typescript
+await revokeOtherSessions();
+await revokeSession(auth.currentSessionID);
```

Comment:
I prefer we disallow to revoke current session. Instead, the developer should `logout`.

Reply:
Revoking a session and logging out can have different meaning, and I see no technical reason to disallow it.

Reply:
Agree, revoke and logout is two things.

Do we need a short cut for revokeAllSessions?

Reply:
Not much application do revoke all sessions including self, most are either revoke individually or all other sessions. So I think there's no need to support it natively.

Reply:
> Revoking a session and logging out can have different meaning, and I see no technical reason to disallow it.

I agree they are two things. What I want to emphasize is when the developer would want to `revokeSession(auth.currentSessionID)` instead of `logout()`. Even if revoking the current session is technically possible, it does not mean it is sensible.

> Do we need a short cut for revokeAllSessions?

That would log the user out. I think the user case is very rare.

Reply:
At the moment, I can't come up with a use case involving revoking current session. We could disallow revoking current session for now if it does not make sense.

Reply:
Good.